### PR TITLE
Exercise 1

### PR DESCRIPTION
**Pull request by AmyNguyen1130**
@Phuongnguyen Please consider my first exercise

"Description"
Your task is to sort a given string. Each word in the string will contain a single number. This number is the position the word should have in the result.

Note: Numbers can be from 1 to 9. So 1 will be the first word (not 0).

If the input string is empty, return an empty string. The words in the input String will only contain valid consecutive numbers.

Examples
"is2 Thi1s T4est 3a"  -->  "Thi1s is2 3a T4est"
"4of Fo1r pe6ople g3ood th5e the2"  -->  "Fo1r the2 g3ood 4of th5e pe6ople"

### CODE REVIEW

File: src/codeenginestudio/huong/Exercise1.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            return false;
+        }
+    }
+
+    public static char[] convertToArray (String str){
+        char[] charArray = str.toCharArray();
+        return charArray;
+    }
+
+    public static void sortString(){
```
Comment (phuongnguyences) [inline src/codeenginestudio/huong/Exercise1.java]: String[] should pass from method invoker.

File: src/codeenginestudio/huong/Exercise1.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        int max = -1;
+        for (int i = 0; i < strArray.length; i++){
+            char[] itemToArray = convertToArray(strArray[i]);
+            for(int j = 0; j <= itemToArray.length -1; j++){
+                if(isNumeric(String.valueOf(itemToArray[j])) == true){
+                    if(itemToArray[j] < max){
+                        temp = strArray[i];
+                        strArray[i] = strArray[i - 1];
+                        strArray[i - 1] = temp;
+                        i -= 2;
```
Comment (haubuicodeengine) [inline src/codeenginestudio/huong/Exercise1.java]: Why i = i - 2 in here ?

File: src/codeenginestudio/huong/Exercise1.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package codeenginestudio.huong;
+
+import java.util.Arrays;
+
+public class Exercise1 {
+    public static void main(String[] args) {
+
+        String[] strArray = {"is2", "pa6" ,"Thi1s", "T4est", "3a", "th5"};
```
Comment (phuongnguyences) [inline src/codeenginestudio/huong/Exercise1.java]: As I read from the requirement, they want to make a function with input is a String, and output also is a String.
IMO, we should have a method with parameter is a String and return type is a String. It makes easy to understand what we need to do.

File: src/codeenginestudio/huong/Exercise1.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            Integer.parseInt(str);
+            return true;
+        } catch(NumberFormatException e){
+            return false;
+        }
+    }
+
+    public static char[] convertToArray (String str){
+        char[] charArray = str.toCharArray();
+        return charArray;
```
Comment (phuongnguyences) [inline src/codeenginestudio/huong/Exercise1.java]: In this function, the variable charArray used to return. Pls, use return str.toCharArray() for instead!

File: src/codeenginestudio/huong/Exercise1.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public static void sortString(String[] strArray){
+
+        String temp = "";
+        int max = -1;
+        for (int i = 0; i < strArray.length; i++){
+            char[] itemToArray = convertToArray(strArray[i]);
+            for(int j = 0; j <= itemToArray.length -1; j++){
+                if(isNumeric(String.valueOf(itemToArray[j])) == true){
```
Comment (phuongnguyences) [inline src/codeenginestudio/huong/Exercise1.java]: if condition valid with 2 returns are true and false. So we don't need to compare it with true here.

File: src/codeenginestudio/huong/Exercise1.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                    if(itemToArray[j] < max){
+                        temp = strArray[i];
+                        strArray[i] = strArray[i - 1];
+                        strArray[i - 1] = temp;
```
Comment (phuongnguyences) [inline src/codeenginestudio/huong/Exercise1.java]: Could you write to a separate method to switch elements in the array?

File: src/codeenginestudio/huong/Exercise1.java (hunk 3 above)
Comment (phuongnguyences) [inline src/codeenginestudio/huong/Exercise1.java]: They will give us a String as an input! Please write a method to slice a string into an array!

**Review by phuongnguyences — COMMENTED** [review]
@AmyNguyen1130 I have some comments, please double-check and let me know if you have any concern!
Thanks.

**Pull request closed without merging**